### PR TITLE
Add a popup for when video render is done

### PR DESCRIPTION
**Pull request by furo321**
image

This should hopefully make it less confusing about where the videos are saved once it's done rendering.

## Checklist

- [x] Tested the change ingame
- [x] Provided screenshots if it is a visual change
- [ ] Tested in combination with possibly related configuration options
- [ ] Written a unit test (especially base/) or added coverage to integration test
- [ ] Considered possible null pointers and out of bounds array indexing
- [ ] Changed no physics that affect existing maps
- [ ] Tested the change with [ASan+UBSan or valgrind's memcheck](https://github.com/ddnet/ddnet/#using-addresssanitizer--undefinedbehavioursanitizer-or-valgrinds-memcheck) (optional)

### CODE REVIEW

File: src/game/client/components/menus.cpp [hunk 1]
```diff
@@ -1677,6 +1681,50 @@ int CMenus::Render()
...
+			{
+				m_Popup = POPUP_NONE;
+				m_DemoRenderInput.Clear();
+			}
+
+			Box.HSplitBottom(160.f, &Box, &Part);
+
+			Part.VSplitLeft(60.0f, 0, &TextBox);
+			TextBox.VSplitLeft(20.0f, 0, &TextBox);
+			TextBox.VSplitRight(60.0f, &TextBox, 0);
```
Comment (Robyt3) [inline src/game/client/components/menus.cpp:1722]: Use `nullptr` instead of `0`

File: src/game/client/components/menus.cpp [hunk 2]
```diff
@@ -1677,6 +1681,50 @@ int CMenus::Render()
...
+
+			str_format(aBuf, sizeof(aBuf), Localize("Video was saved to '%s'"), aFilePath);
+
+			UI()->DoLabel(&TextBox, aBuf, 18.0f, TEXTALIGN_LEFT);
```
Comment (Robyt3) [inline src/game/client/components/menus.cpp:1726]: Use `TEXTALIGN_TL` or `TEXTALIGN_ML`.

Alignment also needs to be tested with very long file names.

File: src/game/client/components/menus_demo.cpp [hunk 3]
```diff
@@ -986,6 +986,13 @@ void CMenus::RenderDemoList(CUIRect MainView)
 		s_Inited = 1;
 	}
 
+#if defined(CONF_VIDEORECORDER)
+	if(m_DemoRenderInput.GetLength() > 0)
```
Comment (Robyt3) [inline src/game/client/components/menus_demo.cpp:990]: Use `!IsEmpty()` instead

**Review by Robyt3 — CHANGES_REQUESTED** [review]

**Comment by Jupeyy** [comment]
It makes it less confusing to know where the videos are stored.

But once you know it, it makes it more annoying to use the render function

**Comment by furo321** [comment]
That's true, but I don't think it's that big of an issue. Also even if you already know the path, you can now just tap a button to go there instead of navigating through a maze of folders.

**Comment by Robyt3** [comment]
Maybe only put a button to open the videos folder in the demo browser?

screenshot

**Comment by furo321** [comment]
There isn't a lot of space to put it, so I halved the size of the demos button. I personally think it looks pretty cramped, what do you think?
image

**Comment by Robyt3** [comment]
To be honest, the demo browser UI should probably be reworked in any case. Some or all buttons could use icons instead of text to save space.

**Comment by Teero888** [comment]
> But once you know it, it makes it more annoying to use the render function (esp. if u render smaller clips and not a huge demo)

I think a little checkbox like "do not show again" would be an idea.

**Review by def- — APPROVED** [review]

**Comment by Jupeyy** [comment]
> Maybe only put a button to open the videos folder in the demo browser?
> 
> screenshot

Wasn't the plan rather to do this, instead of this extra click?

**Comment by Robyt3** [comment]
I didn't find a great spot for a "Videos directory" button in #7311. Maybe the "Demos directory" button could be slightly smaller and moved to the left.

**Comment by Jupeyy** [comment]
Yeah my point is more: If ppl find the render button. they also find the movie button ^^ (maybe even directly next to the render button)

Then we don't need this _ugly_ extra popup.

Except you say it's very user friendly

**Pull request merged**